interpreter/cling/lib/Interpreter/IncrementalParser.cpp

```diff
@@ -201,16 +242,26 @@ namespace cling {
       return;
     }
 
+    std::vector<std::unique_ptr<ASTConsumer>> PluginConsumers;
+    HandlePlugins(*m_CI, PluginConsumers);
+
     DiagnosticsEngine& Diag = m_CI->getDiagnostics();
     if (m_CI->getFrontendOpts().ProgramAction != frontend::ParseSyntaxOnly) {
-      m_CodeGen.reset(CreateLLVMCodeGen(
+      m_CodeGen = CreateLLVMCodeGen(
           Diag, makeModuleName(), m_CI->getHeaderSearchOpts(),
           m_CI->getPreprocessorOpts(), m_CI->getCodeGenOpts(),
-          *m_Interpreter->getLLVMContext()));
+          *m_Interpreter->getLLVMContext());
+      // FIXME: This is suboptimal if PluginConsumers.size() is 0. Clang API
+      // forces us to create a MultiplexConsumer for CodeGen only because of
+      // ownership. m_CodeGen is owned by the multiplexer if there is a plugin
+      // and disallowing us to make m_CodeGen a unique_ptr.
+      auto CG = std::unique_ptr<clang::CodeGenerator>(m_CodeGen);
+      PluginConsumers.push_back(std::move(CG));
     }
 
+    m_PluginConsumer.reset(new MultiplexConsumer(std::move(PluginConsumers)));
     // Initialize the DeclCollector and add callbacks keeping track of macros.
-    m_Consumer->Setup(this, m_CodeGen.get(), m_CI->getPreprocessor());
+    m_Consumer->Setup(this, m_PluginConsumer.get(), m_CI->getPreprocessor());
```

Comment:
The roles of `m_Consumer` versus `m_PluginConsumer` is all but clear; I bet it's a naming issue?